adapters/repos/db/index.go

```diff
@@ -161,6 +161,15 @@ func (i *Index) IterateObjects(ctx context.Context, cb func(index *Index, shard
 	return nil
 }
 
+func (i *Index) IterateShards(ctx context.Context, cb func(index *Index, shard *Shard) error) error {
```

Comment:
IterateShards() does iterate but additionally it might change the state of index or shard.
I would try to document the expected behaviour and whether it is ok for `cb` to mutate the  index